### PR TITLE
Example performance

### PR DESCRIPTION
**Pull request by denisshevchenko**
description
-----------

- [x] `example-performance` for time measurements (see `README.md`).

checklist
---------

- [x] compiles (`cabal new-clean; cabal new-build`)
- [x] tests run successfully (`cabal new-test`)
- [x] documentation added and created (`cd docs; nix-shell --run make`)
- [ ] link to an issue
- [x] link to an epic
- [x] add estimate points
- [x] add milestone (the same as the linked issue)

### CODE REVIEW

File: iohk-monitoring/examples/performance/Main.lhs [hunk 1]
```diff
@@ -0,0 +1,92 @@
+\subsubsection{Module header and import directives}
+\begin{code}
+{-# LANGUAGE CPP                 #-}
+{-# LANGUAGE ScopedTypeVariables #-}
+
+#if defined(linux_HOST_OS)
+#define LINUX
```
Comment (CodiePP) [inline iohk-monitoring/examples/performance/Main.lhs]: not used. and then, `CPP` can be removed as well.

File: iohk-monitoring/iohk-monitoring.cabal [hunk 2]
```diff
@@ -344,3 +344,27 @@ executable example-complex
...
+                       criterion,
+                       text,
+                       unordered-containers
+  if os(windows)
+     build-depends:    Win32
+  else
+     build-depends:    unix
+
+  if os(linux)
+     build-depends:    download
```
Comment (CodiePP) [inline iohk-monitoring/iohk-monitoring.cabal]: why is this needed?

File: iohk-monitoring/src/Cardano/BM/Backend/Switchboard.lhs [hunk 3]
```diff
@@ -141,11 +142,7 @@ The queue is initialized and the message dispatcher launched.
 instance IsEffectuator Switchboard a where
     effectuate switchboard item = do
         let writequeue :: TBQ.TBQueue (LogObject a) -> LogObject a -> IO ()
-            writequeue q i = do
-                    nocapacity <- atomically $ TBQ.isFullTBQueue q
-                    if nocapacity
-                    then handleOverflow switchboard
-                    else atomically $ TBQ.writeTBQueue q i
```
Comment (CodiePP) [inline iohk-monitoring/src/Cardano/BM/Backend/Switchboard.lhs]: we do not want to change this for production. enclosed in flag?

File: iohk-monitoring/src/Cardano/BM/Backend/Switchboard.lhs [hunk 4]
```diff
@@ -249,7 +246,7 @@ instance (FromJSON a, ToObject a) => IsBackend Switchboard a where
 
                 Async.async $ qProc countersMVar
 
-        q <- atomically $ TBQ.newTBQueue 2048
+        q <- atomically $ TBQ.newTBQueue 1000000
```
Comment (CodiePP) [inline iohk-monitoring/src/Cardano/BM/Backend/Switchboard.lhs]: we do not want to have such a big queue in production. chould be conditional on a flag?

Comment (denisshevchenko) [reply iohk-monitoring/src/Cardano/BM/Backend/Switchboard.lhs]: Done.
